### ISSUE TITLE
save files to shared drive

### ISSUE THREAD

**Issue by francescolosterzo** [ahochsteger/gmail2gdrive#34]
Hello!

thanks a lot for the nice work!

In my organization we use Google services and we use [Shared Drives](https://gsuite.google.com/intl/en_ie/products/drive/?utm_source=google&utm_medium=cpc&utm_campaign=emea-ch-all-en-dr-bkws-all-lv-trial-e-t4-1008069&utm_content=text-ad-none-none-DEV_c-CRE_251889814519-ADGP_Desk%2BTab%20%7C%20AW%20SEM%20%7C%20BKWS%20~%20EXA_1:1_CH_EN_Drive_share%20google%20drive_Gap%20Analysis%2026.01.2018-KWID_43700030414876631-kwd-299031766319-userloc_1002998&utm_term=KW_share%20google%20drive-ST_share%20google%20drive&ds_rl=1259922&gclid=EAIaIQobChMIoM3y0ICG6AIVVuDtCh1U8ALiEAAYASAAEgJlP_D_BwE&gclsrc=aw.ds). I was wondering if there is the possibility to save the files (e.g. attachments) in a folder in a Shared Drive instead of a folder in `MyDrive` ?

I guess some modification would be needed in `function getFolderByPath(path)` where by default it starts from the root.

Do you have any suggestions?

**Comment by Wanders77** [comment]
Hi I think this will work for you, it has so far for me. 

Replace DriveApp.getRootFolder() in the functions getFolderByPath and getOrCreateFolder with DriveApp.getFolderById("FOLDER ID") you can pull the folder id from the drive URL it is the set of characters after drive/folders/

That will set the default path to your selected shared folder.

**Comment by francescolosterzo** [comment]
Hi!

Thanks a lot for the suggestion! It worked in the end, here are some details:

> Replace DriveApp.getRootFolder() in the functions getFolderByPath and getOrCreateFolder with DriveApp.getFolderById("FOLDER ID")

I tried to do this but I was getting errors from inside `getFolderByPath()`, coming from the fact that it loops through the subfolders in the full path (i.e. the outcome of `path.split("/");`), but the folder id taken from the URL is not a path.

Instead of the above, I had to replace **the whole** `getFolderByPath()` with `DriveApp.getFolderById("FOLDER ID")` and it works perfectly!

**Comment by francescolosterzo** [comment]
👆 this is solved

**Comment by sshah23** [comment]
Hello @francescolosterzo :)

Could you kindly share the whole Code.js file, with the same code change you made above?

Regards.

**Comment by francescolosterzo** [comment]
hey @sshah23 
it is very simple indeed!
previous code:
```
var folder = getFolderByPath(googleDriveFolder);
var file = folder.createFile(attachment);
```
in which `googleDriveFolder` is the complete path to the folder, starting from MyDrive. E.g. if in MyDrive I have "dir1/dir2/" and I want to save the file in "dir2" I'd have:
```
googleDriveFolder = "dir1/dir2;
```

new code:
```
var folder = DriveApp.getFolderById(googleDriveFolderID)
var file = folder.createFile(attachment);
```
where `googleDriveFolderId` is the id of the folder, i.e. the string that comes after `https://drive.google.com/drive/folders/` in the url of the folder.

**Comment by sshah23** [comment]
@francescolosterzo I appreciate your explanation, but I am using the default config as of here:
https://github.com/ahochsteger/gmail2gdrive/blob/master/Code.gs

So in your previous comment before, you mentioned:

> Instead of the above, I had to replace the whole getFolderByPath() with DriveApp.getFolderById("FOLDER ID") and it works perfectly!

Line 42 to 57 is that function:
https://github.com/ahochsteger/gmail2gdrive/blob/23d55de4306ef984a60e35d7dc44290e9b6a6f53/Code.gs#L42

On Line 47, it says:
`var folder = DriveApp.getRootFolder();`

Shall I replace that with:
`var folder = DriveApp.getFolderById("FOLDER ID")`

Where FOLDER ID is the actual Folder ID

Line 94:
https://github.com/ahochsteger/gmail2gdrive/blob/master/Code.gs#L94

Has  `var file = folder.createFile(attachment);` - that should be replaced with `var file = folder.createFile(attachment);`

Apologise if it's simple, but I'm a slight beginner

**Comment by francescolosterzo** [comment]
@sshah23 no worries!
my code is a bit different than the original one, but is't much simpler as I had to deal with a very specific problem so I didn't need all the flexibility that the original project has. Because of this in my code I called `getFolderPath()` directly from inside the "main" function instead of having it nested inside another function. This might have caused some confusion :)

In short I replaced [line 93](https://github.com/ahochsteger/gmail2gdrive/blob/master/Code.gs#L93) with the simple expression:
```
var folder = DriveApp.getFolderById(googleDriveFolderID)
```
`googleDriveFolderID` being the ID of the folder. The rest stays the same.

**Comment by sshah23** [comment]
@francescolosterzo works perfectly! Thank you so much :).

**Comment by sshah23** [comment]
@francescolosterzo Hey buddy. So from my testing, it seems to run perfectly well when I manually run the `Gmail2Drive` Function in the actual Script Editor (retrieves the files via email and uploads them successfully to the Shared Google Drive folder; which I've defined via the ID of the folder), but when I set up a Trigger to execute `Gmail2Drive`:

![My_Triggers_-_Apps_Script](https://user-images.githubusercontent.com/17063034/84753869-7977ed00-afb7-11ea-99b8-8068d0929319.png)

It runs, but the file doesn't get uploaded to the Shared folder. Instead, the same folder structure is created on the root of my Google Drive (same folder names / path / structure as the existing Shared folder), but it's a brand new set of folders.

Any ideas?

**Comment by francescolosterzo** [comment]
@sshah23 not really, as I never had to deal with the triggers (the project I was working on when dealing with this finished quickly and therefore I didn't have to bother in setting up a trigger). Sorry :\

**Comment by sshah23** [comment]
@francescolosterzo No worries at all :) I'll have a play around and see what it might be. Thank you for your help though 🙌

**Comment by wsolorza** [comment]
I keep getting this error.

https://a.cl.ly/E0u911ZP

Tried both below

 
@sshah23 @francescolosterzo 
Instead of the above, I had to replace the whole getFolderByPath() with DriveApp.getFolderById("FOLDER ID") and it works perfectly!

@sshah23 no worries!
my code is a bit different than the original one, but is't much simpler as I had to deal with a very specific problem so I didn't need all the flexibility that the original project has. Because of this in my code I called getFolderPath() directly from inside the "main" function instead of having it nested inside another function. This might have caused some confusion :)

In short I replaced line 93 with the simple expression:

var folder = DriveApp.getFolderById(googleDriveFolderID)
